src/curve_fitting.py

```diff
@@ -0,0 +1,272 @@
+                    break
+                param_entries = param_entries.get_parent()
+
+            # Set the parameters for the row corresponding to the entry that
+            # was edited
+            if row == param_entries:
+                initial = param_entries.initial.get_text()
+                lower_bound = param_entries.lower_bound.get_text()
+                upper_bound = param_entries.upper_bound.get_text()
+
+                self.fitting_parameters[index].initial = (
+                    initial if is_float(initial) else 1)
+                self.fitting_parameters[index].lower_bound = (
+                    lower_bound if is_float(lower_bound) else -float("inf"))
+                self.fitting_parameters[index].upper_bound = (
+                    upper_bound if is_float(upper_bound) else float("inf"))
+        self.fit_curve()
+
+    def set_results(self):
+        initial_string = _("Results: \n")
+        buffer_string = initial_string
+        for index, arg in enumerate(self.get_free_variables()):
+            buffer_string += f"\n {arg}: {self.param[index]}"
+
+        self.text_view.get_buffer().set_text(buffer_string)
+        bold_tag = Gtk.TextTag(weight=700)
+        self.text_view.get_buffer().get_tag_table().add(bold_tag)
+
+        start_iter = self.text_view.get_buffer().get_start_iter()
+        end_iter = self.text_view.get_buffer().get_start_iter()
+
+        # Highlight first word
+        while not end_iter.ends_word() and not end_iter.ends_sentence():
+            end_iter.forward_char()
+        self.text_view.get_buffer().apply_tag(bold_tag, start_iter, end_iter)
+
+    @property
+    def equation(self):
+        return preprocess(str(self.equation_entry.get_text()))
+
+    def fit_curve(self):
+        def _get_equation_name(equation_name, values):
+            var_to_val = dict(zip(self.get_free_variables(), values))
+
+            for var, val in var_to_val.items():
+                equation_name = equation_name.replace(var, str(round(val, 3)))
+            return equation_name
+
+        function = string_to_function(self.equation)
+        if function is None:
+            return
+        try:
+            self.param, param_cov = \
+                curve_fit(function, self.curves[1].xdata,
+                          self.curves[1].ydata,
+                          p0=self.fitting_parameters.get_p0(),
+                          bounds=self.fitting_parameters.get_bounds(),
+                          nan_policy="omit")
+        except ValueError:
+            # Cancel fit if not succesful
+            return
+        xdata = numpy.linspace(
+            min(self.curves[1].xdata),
+            max(self.curves[1].xdata),
+            5000)
+        ydata = [function(x, *self.param) for x in xdata]
+
+        name = _get_equation_name(
+            str(self.equation_entry.get_text()), self.param)
+        self.curves[0].name = f"Y = {name}"
+        self.curves[0].ydata, self.curves[0].xdata = (ydata, xdata)
```

Comment:
this would just mean adding a canvas.set_items(curves) here

Reply:
So I've been trying this as a way to update the canvas upon fits, but without any success. There's no refresh being triggered.